piptools/utils.py

```diff
@@ -88,6 +93,22 @@ def comment(text: str) -> str:
     return click.style(text, fg="green")
 
 
+def canonicalize_ireq(ireq: InstallRequirement) -> InstallRequirement:
+    """
+    Return a copy of ireq with canonicalized extras strings
+    """
+    ireq = copy_install_requirement(
+        ireq, extras=set(map(canonicalize_name, ireq.extras))
+    )
+    if ireq.req:
+        ireq.req.extras = set(ireq.extras)
```

Comment:
Did you mean
```suggestion
        ireq.req.extras = set(map(canonicalize_name, ireq.extras))
```
?

Reply:
No, because they get normalized on line 101.